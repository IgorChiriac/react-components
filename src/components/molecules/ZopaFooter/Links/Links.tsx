import React from 'react';
import styled from 'styled-components';
import * as colors from '../../../../constants/colors';
import FlexCol from '../../../layout/FlexCol/FlexCol';
import FlexRow from '../../../layout/FlexRow/FlexRow';
import FooterLink from '../FooterLink/FooterLink';
import Separator from '../Separator/Separator';
import Heading from '../../../atoms/Heading/Heading';
import Wrapper from '../Wrapper/Wrapper';

export interface ILinkData {
  label: string;
  href: string;
}

export interface ILinkGroups {
  heading: string;
  links: ILinkData[];
}

const StyledHeading = styled(Heading)`
  color: ${colors.base.white};
`;

const StyledList = styled.ul`
  margin: 0;
  padding: 0;
  list-style-type: none;
`;

const StyledListItem = styled.li`
  margin: 0 0 8px 0;
  padding: 0;
`;

const linkGroups = [
  {
    heading: 'About Zopa',
    links: [
      {
        href: 'https://zopa.com/about',
        label: 'About Zopa',
      },
      {
        href: 'https://zopa.com/about/how-zopa-works',
        label: 'How Zopa works',
      },
      {
        href: 'https://zopa.com/about/join-us',
        label: 'Join us',
      },
      {
        href: 'https://zopa.com/about/press',
        label: 'Press office',
      },
      {
        href: 'https://zopa.com/contact',
        label: 'Contact Zopa',
      },
    ],
  },
  {
    heading: 'Zopa Loans',
    links: [
      {
        href: 'https://zopa.com/loans/debt-consolidation',
        label: 'Debt consolidation',
      },
      {
        href: 'https://zopa.com/loans/home-improvement',
        label: 'Home improvements',
      },
      {
        href: 'https://zopa.com/loans/car-loans',
        label: 'Car loans',
      },
      {
        href: 'https://zopa.com/loans/faq',
        label: 'FAQs for loans',
      },
    ],
  },
  {
    heading: 'Invest your money',
    links: [
      {
        href: 'https://zopa.com/lending',
        label: 'Invest at Zopa',
      },
      {
        href: 'https://zopa.com/lending/risk-management',
        label: 'Risk management',
      },
      {
        href: 'https://zopa.com/lending/faq',
        label: 'FAQs for investing',
      },
    ],
  },
];

const Links = () => (
  <FlexRow>
    {linkGroups.map(({ heading, links }, index) => (
      <FlexCol xs={12} s={6} m={4} l={3} key={`footer-links-${index}`}>
        <Wrapper>
          <StyledHeading>{heading}</StyledHeading>
          <StyledList>
            {links.map(({ label, href }) => (
              <StyledListItem key={label}>
                <FooterLink href={href}>{label}</FooterLink>
              </StyledListItem>
            ))}
          </StyledList>
        </Wrapper>
        <FlexCol xs="fill" s="hidden">
          <Separator />
        </FlexCol>
      </FlexCol>
    ))}
  </FlexRow>
);

export default Links;
