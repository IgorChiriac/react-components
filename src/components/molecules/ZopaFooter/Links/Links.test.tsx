import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import Links from './Links';

describe('<Links />', () => {
  it('renders the component', async () => {
    const { container } = render(<Links />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with no a11y violations', async () => {
    const { container } = render(<Links />);
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
