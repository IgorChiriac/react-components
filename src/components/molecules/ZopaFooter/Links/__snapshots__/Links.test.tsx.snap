// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> renders the component 1`] = `
.c1 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 16px;
  padding-left: 16px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c8 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: -16px;
  margin-left: -16px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c7 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #4A3EDE;
}

.c7:hover {
  opacity: 0.88;
}

.c7:active {
  opacity: 0.72;
}

.c6 {
  color: #A9ACBA;
}

.c6:hover {
  color: #6F748B;
}

.c9 {
  background-color: #282F52;
  height: 1px;
}

.c2 {
  padding: 18px 0;
}

.c3 {
  color: #FFFFFF;
}

.c4 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c5 {
  margin: 0 0 8px 0;
  padding: 0;
}

@media (min-width:0px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 100%;
    -ms-flex: 0 0 100%;
    flex: 0 0 100%;
    max-width: 100%;
  }
}

@media (min-width:576px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 50%;
    -ms-flex: 0 0 50%;
    flex: 0 0 50%;
    max-width: 50%;
  }
}

@media (min-width:768px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 33.33333333333333%;
    -ms-flex: 0 0 33.33333333333333%;
    flex: 0 0 33.33333333333333%;
    max-width: 33.33333333333333%;
  }
}

@media (min-width:992px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 25%;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    max-width: 25%;
  }
}

@media (min-width:0px) {
  .c8 {
    display: block;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

@media (min-width:576px) {
  .c8 {
    display: none;
  }
}

<div
  class="c0"
  cols="12"
  direction="row"
>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <h3
        class="c3"
      >
        About Zopa
      </h3>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/about"
          >
            About Zopa
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/about/how-zopa-works"
          >
            How Zopa works
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/about/join-us"
          >
            Join us
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/about/press"
          >
            Press office
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/contact"
          >
            Contact Zopa
          </a>
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      />
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <h3
        class="c3"
      >
        Zopa Loans
      </h3>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/loans/debt-consolidation"
          >
            Debt consolidation
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/loans/home-improvement"
          >
            Home improvements
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/loans/car-loans"
          >
            Car loans
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/loans/faq"
          >
            FAQs for loans
          </a>
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      />
    </div>
  </div>
  <div
    class="c1"
    cols="12"
  >
    <div
      class="c2"
    >
      <h3
        class="c3"
      >
        Invest your money
      </h3>
      <ul
        class="c4"
      >
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/lending"
          >
            Invest at Zopa
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/lending/risk-management"
          >
            Risk management
          </a>
        </li>
        <li
          class="c5"
        >
          <a
            class="c6 c7"
            href="https://zopa.com/lending/faq"
          >
            FAQs for investing
          </a>
        </li>
      </ul>
    </div>
    <div
      class="c8"
    >
      <div
        class="c9"
      />
    </div>
  </div>
</div>
`;
